### ISSUE TITLE
Inference slowed down by pruning and quantization

### ISSUE THREAD

**Issue by vvolhejn** [neuralmagic/deepsparse#485]
Hi, I'm trying to use DeepSparse to run 1D CNNs for audio processing. I was benchmarking the performance depending on whether or not pruning (with 0.9 sparsity) and/or quantization is used, i.e. four different models. I'm using an AVX512 CPU (no VNNI).

I would expect the ordering of performance (higher is better) to be
- vanilla < quantized <= quantized+pruned
- vanilla < pruned <= quantized+pruned

If I understand correctly, quantized+pruned might not improve performance because "sparse quantization" only works with VNNI as stated [here](https://docs.neuralmagic.com/deepsparse/main/index.html).

However, the inference times I measured for the models are:
- vanilla: 0.029604
- quantized: 0.03177
- pruned+quantized: 0.096895
- pruned: 0.13405
which does not correspond to the expectations. I got these times using my own custom benchmark, but using `deepsparse.benchmark path-to-model.onnx -nstreams 1 -s sync -b 1` confirmed the times as well.

[Here](https://drive.google.com/drive/folders/1G5_z0m1XFKRrP9uh0WVJXx9WgGrwROy1?usp=sharing) are .onnx files of the models I was using -- I hope the naming scheme is clear.

I pruned/quantized the models using SparseML, where I built the recipes using these templates:
```python
base_recipe_template = """
version: 0.1.0
modifiers:
    - !EpochRangeModifier
        start_epoch: 0.0
        end_epoch: 2

    - !LearningRateModifier
        start_epoch: 0
        end_epoch: 2
        init_lr: 0.005
        lr_class:  ExponentialLR
        lr_kwargs:
            gamma: 0.9
"""

pruning_template = """
    - !GMPruningModifier
        start_epoch: 0
        end_epoch: 1
        update_frequency: 1.0
        init_sparsity: 0.05
        final_sparsity: 0.9
        mask_type: block4
        params: {layers_to_prune}
"""

quantization_template = """
    - !QuantizationModifier
        start_epoch: 0.0
"""
```
where `layers_to_prune = sparseml.pytorch.utils.get_prunable_layers(model)`. I only add `pruning_template` for the pruned models and `quantization_template` for the quantized models.

I initialize the model randomly and train on random data (`torch.randn_like(...)`).

**Expected behavior**: quantization and pruning should decrease inference time, not increase it.

**Environment**
Include all relevant environment information:
1. OS [e.g. Ubuntu 18.04]: Debian GNU/Linux 11 (bullseye)
2. Python version [e.g. 3.7]: 3.8.12
3. DeepSparse version or commit hash [e.g. 0.1.0, `f7245c8`]: 0.12.2
4. ML framework version(s) [e.g. torch 1.7.1]: Torch 1.9.1
5. Other Python package versions [e.g. SparseML, Sparsify, numpy, ONNX]: SparseML 0.12.2, ONNX 1.10.1, 
6. CPU info - output of deepsparse/src/deepsparse/arch.bin or output of cpu_architecture(): `{'vendor': 'GenuineIntel', 'isa': 'avx512', 'vnni': False, 'num_sockets': 1, 'available_sockets': 1, 'cores_per_socket': 1, 'available_cores_per_socket': 1, 'threads_per_core': 2, 'available_threads_per_core': 2, 'L1_instruction_cache_size': 32768, 'L1_data_cache_size': 32768, 'L2_cache_size': 1048576, 'L3_cache_size': 40370176}`

**To Reproduce**: Run [the .onnx files](https://drive.google.com/drive/folders/1G5_z0m1XFKRrP9uh0WVJXx9WgGrwROy1?usp=sharing) (with batch size 1) and measure latency, e.g. using `deepsparse.benchmark path-to-model.onnx -nstreams 1 -s sync -b 1`

Am I doing something wrong? Why is this happening?

**Comment by tlrmchlsmth** [comment]
@vvolhejn that's right on the dilated convolutions -- we don't support them for performance and have to fall back to a different implementation.

As for your original 1D CNNs: for the sparse case, deepsparse currently does better if the last axis does not have length 1. E.g. if your input were `1x1x1x64000` instead of `1x1x64000x1`. This is not ideal, so I'm working on a fix for this and can let you know when it's in the nightly release.

**Comment by tlrmchlsmth** [comment]
Hi @vvolhejn you should see an overall performance increase on the current `deepsparse-nightly`. For batch size 1, the sparse models are quite a bit faster than before.
